### PR TITLE
Fix for The ArrayObject Bug in the CachedClient

### LINKED ISSUE #253: ArrayObject-Bug in Jackalope\Transport\DoctrineDBAL\Client.php
Since namespace is an ArrayObject now, there is the need to cast it to an array...which is done in:

   /**
     * {@inheritDoc}
     */
    public function getNamespaces()
    {
        return (array) $this->getNamespacesObject();
    }

But this is overriden by the CachedClient "getNamespaces"-Method.

Which just checks if namespaces is filled:

 /**
     * {@inheritDoc}
     */
    public function getNamespaces()
    {
        if (empty($this->namespaces)) {
            $cacheKey = 'namespaces';
            $result = $this->caches['meta']->fetch($cacheKey);
            if ($result) {
                $this->namespaces = $result;
            } else {
                $this->namespaces = parent::getNamespaces();
                $this->caches['meta']->save($cacheKey, $this->namespaces);
            }
        }
        return $this->namespaces;
    }

And if so, it justs returns its...but it is not casted there....so the engine throws a warning.

Easy fix is to cast it in the CachedClient as well...but a less "doubled" solution would be better i think. :)

Like casting it in "getNamespacePrefixes()" in the Client-Class.

### PR DESCRIPTION
**Pull request by ARoddis**
I had to change the visibility of the "getNamespacesObject" Method. I hope that is ok.

fix #253

### CODE REVIEW

**Comment by dbu** [comment]
looks good to me. visiblity change is ok, we default to private but when there is a good reason protected is fine.

@dantleech can you check this as well please?

**Comment by lsmith77** [comment]
ping

File: tests/Jackalope/Transport/DoctrineDBAL/CachedClientTest.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?php
+namespace Jackalope\Transport\DoctrineDBAL;
+
+use Doctrine\DBAL\Platforms\SqlitePlatform;
+use Jackalope\Test\TestCase;
+use PHPCR\PropertyType;
+use PHPCR\Util\NodeHelper;
+use PHPCR\Util\PathHelper;
+
+
```
Comment (dantleech) [inline tests/Jackalope/Transport/DoctrineDBAL/CachedClientTest.php]: remove extra line, missing header.

File: src/Jackalope/Transport/DoctrineDBAL/Client.php [hunk 2]
```diff
@@ -522,7 +522,7 @@ public function getNamespaces()
      *
      * @return \ArrayObject
      */
-    private function getNamespacesObject()
+    protected function getNamespacesObject()
```
Comment (dantleech) [inline src/Jackalope/Transport/DoctrineDBAL/Client.php]: Does this need to be `protected`?

File: tests/Jackalope/Transport/DoctrineDBAL/CachedClientTest.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    private $repository;
+
+    /**
+     * @var \Jackalope\Session
+     */
+    private $session;
+
+    public function setUp()
+    {
+        static $initialized = false;
```
Comment (dantleech) [inline tests/Jackalope/Transport/DoctrineDBAL/CachedClientTest.php]: Is this used by anything?

File: tests/Jackalope/Transport/DoctrineDBAL/CachedClientTest.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        $tables = $schema->getTables();
+
+        foreach ($tables as $table) {
+            $conn->exec('DELETE FROM ' . $table->getName());
+        }
+
+        $this->transport = new \Jackalope\Transport\DoctrineDBAL\CachedClient(new \Jackalope\Factory(), $conn);
+        $this->transport->createWorkspace('default');
+
+        $this->repository = new \Jackalope\Repository(null, $this->transport);
```
Comment (dantleech) [inline tests/Jackalope/Transport/DoctrineDBAL/CachedClientTest.php]: @dbu is there nothing existing that does this bootstrapping?

**Comment by dantleech** [comment]
Other than my comment this is good for me.

File: tests/Jackalope/Transport/DoctrineDBAL/CachedClientTest.php (hunk 3 above)
Comment (dbu) [inline tests/Jackalope/Transport/DoctrineDBAL/CachedClientTest.php]: looks like a copy-paste that is also present but not used elsewhere.

File: tests/Jackalope/Transport/DoctrineDBAL/CachedClientTest.php (hunk 4 above)
Comment (dbu) [inline tests/Jackalope/Transport/DoctrineDBAL/CachedClientTest.php]: aparently not. i will do a bit of refactoring of the tests i think.

**Comment by dbu** [comment]
thank you very much @ARoddis - i did the codestyle fixes and opened a new PR which i will merge as soon as its green. i will quickly cleanup the test initialization also.

**Pull request closed without merging**